students/gotem888/hw-12 searchEngine/scripts/scripts.js

```diff
@@ -0,0 +1,417 @@
+            alt: 'Toraverb 2'
+        }
+    },
+    {
+        group_type: 'effects',
+        headline: 'Fav Filter Pro-Q3',
+
+        image: {
+            filename: 'https://i.ibb.co/r50qW4X/Fab-Filter-Pro-Q3.png',
+            alt: 'Pro-Q3'
+        }
+    },
+    {
+        group_type: 'effects',
+        headline: 'HOFA IQ Analyser',
+
+        image: {
+            filename: 'https://i.ibb.co/9hNXZC6/HOFA-IQ-Analyser.png',
+            alt: 'IQ Analyser'
+        }
+    },
+    {
+        group_type: 'effects',
+        headline: 'Image Line Convolver',
+
+        image: {
+            filename: 'https://i.ibb.co/yqbybrs/Image-Line-Convolver.png',
+            alt: 'Convolver'
+        }
+    },
+    {
+        group_type: 'effects',
+        headline: 'Image Line Effector',
+
+        image: {
+            filename: 'https://i.ibb.co/X35pTpd/Image-Line-Effector.png',
+            alt: 'Effector'
+        }
+    },
+    {
+        group_type: 'effects',
+        headline: 'Lifeline Expanse',
+
+        image: {
+            filename: 'https://i.ibb.co/DkRRQS5/Lifeline-Expanse.png',
+            alt: 'Expanse'
+        }
+    },
+    {
+        group_type: 'effects',
+        headline: 'MURDA Melodies',
+
+        image: {
+            filename: 'https://i.ibb.co/rQzn3Y4/MURDA-Melodies.png',
+            alt: 'MURDA Melodies'
+        }
+    }
+];
+
+const createCardContent = (object) => {
+    const {headline, image} = object;
+  
+    const card = `<div class="card">
+      <img class="card__img" src="${image.filename}" alt="${image.alt}">
+      <p class="card__title">${headline}</p>
+      </div>`;
+  
+    return card;
+  };
+  
+  const generateContent = (array) => {
+    const content = document.querySelector('#content');
+  
+    content.innerHTML = '';
+  
+    let data = '';
+  
+    array.forEach((element) => {
+      data += createCardContent(element);
+    });
+  
+    if (!data) data = 'No data';
+  
+    content.innerHTML = data;
+  };
+  
+  const filterData = (data, filter) => {
+    const {vstname, group} = filter;
+    return data.filter((item) => {
+      let result = true;
+  
+      if (vstname) {
+        if (item.headline.toLowerCase().indexOf(vstname.toLowerCase()) === -1) {
+          result = false;
+        }
+      }
+  
+      if (group) {
+        if (item.group_type !== group) {
+          result = false;
+        }
+      }
+  
+      return result;
+    });
+  };
+
+generateContent(PLUGINS);
+
+  document.forms.filtersForm.addEventListener('submit', (e) => {
+    e.preventDefault();
+  
+    const formData = new FormData(e.target);
+    const formProps = Object.fromEntries(formData); // {group: STRING || UNDEFINED, name: STRING || UNDEFINED}
+  
+    const filteredData = filterData(PLUGINS, formProps);
+    generateContent(filteredData);
+  });
+
+
+  document.getElementById('rstButton').addEventListener("click", ()=> {generateContent(PLUGINS)});
```

Comment:
to be more consistent I advise to add listener to `filtersForm` `reset` event
```suggestion
  document.forms.filtersForm.addEventListener("reset", ()=> {generateContent(PLUGINS)});
```